src/nbl/asset/utils/CHLSLCompiler.cpp

```diff
@@ -61,29 +92,105 @@ CHLSLCompiler::~CHLSLCompiler()
 }
 
 
-struct DxcCompilationResult
+static void try_upgrade_hlsl_version(std::vector<std::wstring>& arguments, system::logger_opt_ptr& logger)
 {
-    ComPtr<IDxcBlobEncoding> errorMessages;
-    ComPtr<IDxcBlob> objectBlob;
-    ComPtr<IDxcResult> compileResult;
+    auto stageArgumentPos = std::find(arguments.begin(), arguments.end(), L"-HV");
+    if (stageArgumentPos != arguments.end()) {
+        auto index = stageArgumentPos - arguments.begin() + 1; // -HV XXXXX, get index of second
+        std::wstring version = arguments[index];
+        if (!isalpha(version.back()) && version.length() >= 4 && std::stoi(version) < 2021)
+            arguments[index] = L"2021";
+    }
+    else {
+        logger.log("Compile flag error: Required compile flag not found -HV. Force enabling -HV 202x, as it is required by Nabla.", system::ILogger::ELL_WARNING);
+        arguments.push_back(L"-HV");
+        arguments.push_back(L"202x");
+    }
+}
 
-    std::string GetErrorMessagesString()
+
+static void try_upgrade_shader_stage(std::vector<std::wstring>& arguments, asset::IShader::E_SHADER_STAGE shaderStageOverrideFromPragma, system::logger_opt_ptr& logger) {
+	
+	constexpr int MajorReqVersion = 6,
+		MinorReqVersion = 7;
+
+	auto foundShaderStageArgument = std::find(arguments.begin(), arguments.end(), L"-T");
+	if (foundShaderStageArgument != arguments.end() && foundShaderStageArgument + 1 != arguments.end()) {
+		auto foundShaderStageArgumentValueIdx = foundShaderStageArgument - arguments.begin() + 1;
+		std::wstring s = arguments[foundShaderStageArgumentValueIdx];
+		if (s.length() >= 6) {
+			if (shaderStageOverrideFromPragma != IShader::ESS_UNKNOWN)
+			{
+				// replace first 2 characters if shaderStageOverrideFromPragma != Unknown
+				auto stageStr = ShaderStageToString(shaderStageOverrideFromPragma);
+				if (!stageStr) {
+					logger.log("invalid shader stage %i", system::ILogger::ELL_ERROR, shaderStageOverrideFromPragma);
+					return;
+				}
+				s.replace(0, 2, stageStr);
+			}
+
+			std::vector<std::wstring::iterator> underscorePositions = {};
+			auto it = std::find(s.begin(), s.end(), '_');
+            while (it != s.end()) {
+				underscorePositions.push_back(it);
+                it = std::find(it + 1, s.end(), '_');
+            }
+			std::wstring majorVersionString, minorVersionString;
+			int size = underscorePositions.size();
+			auto secondLastUnderscore = underscorePositions[size - 2];
+			auto lastUnderscore = underscorePositions[size - 1];
+			majorVersionString = std::wstring(secondLastUnderscore + 1, lastUnderscore);
+			minorVersionString = std::wstring(lastUnderscore + 1, s.end());
+
+			int major = std::stoi(majorVersionString);
+			int minor = std::stoi(minorVersionString);
```

Comment:
`stoi` throws `std::invalid_argument` etc. , so these need to be in a try-catch block and the `major` and `minor` need initialization to som einvalid balue